datafusion/physical-expr/src/datetime_expressions.rs

```diff
@@ -1051,6 +1077,91 @@ mod tests {
         });
     }
 
+    #[test]
+    fn test_date_trunc_timezones() {
+        let cases = vec![
+            (
+                vec![
+                    "2020-09-08T00:00:00Z",
```

Comment:
Given the usecase is preseving timezones, it may make sense to also check with timezones other than `Z` (like maybe +5 or +8 🤔

Reply:
I think the test I had for the "-02" zone was sufficient. The way times are stored is always as a UTC offset, the zone information really just indicates how the offset needs to be interpretted. These strings are only used to derive th offset and therefore the specified zone is ignored when creating the array. However, more tests don't hurt so I've added the ones you suggested for clarity.

Reply:
Thanks -- sorry for my confusion